contracts/src/libraries/FeeVault.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/// @title FeeVault
+/// @notice The FeeVault contract contains the basic logic for the various different vault contracts
+///         used to hold fee revenue generated by the L2 system.
+abstract contract FeeVault is Initializable, UUPSUpgradeable, OwnableUpgradeable {
+    /// @notice Minimum balance before a withdrawal can be triggered.
+    uint256 public minWithdrawalAmount;
```

Comment:
do we want this?

Reply:
sure it's fine, just configure it to 0 for now

Reply:
it is configurable in `genesis_config.json` now